src/lib/components/DashNgl.react.js

```diff
@@ -0,0 +1,253 @@
+                this.loadStructure(
+                    stage,
+                    filename,
+                    data[i].chain,
+                    data[i].color,
+                    xOffsetArr[i]
+                );
+            } else {
+                // load from backend
+                this.loadData(data[i], stage, structuresList, xOffsetArr[i]);
+            }
+        }
+        const center = stage.getCenter();
+        const zoom = stage.getZoom();
+        // https://www.youtube.com/watch?v=L8CDt1J3DAw beyond console.log in 100sec
+        console.log({center}, {zoom});
+
+        // stage.autoView()
+        // change zoom depending on sub units
+        const newZoom = -500;
+        const duration = 1000;
+        stage.animationControls.zoomMove(center, newZoom, duration);
+    }
+
+    loadData(data, stage, structuresList, xOffset) {
+        console.log('load from backend');
+        const stringBlob = new Blob([data.config.input], {
+            type: data.config.type,
+        });
+        stage
+            .loadFile(stringBlob, {ext: data.ext, defaultRepresentation: false})
+            .then(stageObj => {
+                stageObj.name = data.filename;
+                this.showStructure(
+                    stageObj,
+                    data.chain,
+                    data.color,
+                    xOffset,
+                    stage
+                );
+
+                this.setState(state => ({
+                    structuresList: state.structuresList.concat([
+                        data.filename,
+                    ]),
+                }));
+                console.log(this.state);
+            });
+    }
+
+    render() {
+        return (
+            <div>
+                <div id="viewport" style={{width: '100%', height: '500px'}} />
+            </div>
+        );
+    }
+}
+
+const defaultViewportStyle = {
+    width: '100%',
+    height: '500px',
+};
+
+const defaultStageParameters = {
+    quality: 'medium',
+    backgroundColor: 'white',
+};
+
+const dataPropShape = {
+    selectedValue: PropTypes.string.isRequired,
+    chain: PropTypes.string.isRequired,
+    color: PropTypes.string.isRequired,
+    filename: PropTypes.string.isRequired,
+    ext: PropTypes.string,
+    config: PropTypes.shape({
+        type: PropTypes.string.isRequired,
+        input: PropTypes.oneOfType([
+            PropTypes.array,
+            PropTypes.object,
+            PropTypes.string,
+        ]),
+    }),
+};
+
+DashNgl.defaultProps = {
+    id: 'viewport',
+    viewportStyle: defaultViewportStyle,
+    stageParameters: defaultStageParameters,
+};
+
+DashNgl.propTypes = {
+    /**
+     * The ID used to identify this component in Dash callbacks.
+     */
+    id: PropTypes.string,
+
+    /**
+     * CSS styling for viewport container
+     */
+    viewportStyle: PropTypes.object,
+
+    /**
+     * Parameters for the stage
+     */
+    stageParameters: PropTypes.object,
```

Comment:
Please be a bit more specific here - this should be a `PropTypes.exact` instead. It would also be great to have comments that are a bit more descriptive and in full sentences, since those are what go in the Python docstring. You can see this component for an example of how it should look: https://github.com/plotly/dash-bio/blob/dcdd7161c3a13790f2b6c6ca26ddad7eed648225/src/lib/components/FornaContainer.react.js#L202